flink-python/pyflink/table/catalog.py

```diff
@@ -1391,3 +1398,422 @@ def of(catalog_name: str, configuration: Configuration, comment: str = None):
         j_catalog_descriptor = gateway.jvm.org.apache.flink.table.catalog.CatalogDescriptor.of(
             catalog_name, configuration._j_configuration, comment)
         return CatalogDescriptor(j_catalog_descriptor)
+
+
+class Column(metaclass=ABCMeta):
+    """
+    Representation of a column in a :class:`~pyflink.table.ResolvedSchema`.
+
+    A table column describes either a :class:`PhysicalColumn`, :class:`ComputedColumn`, or
+    :class:`MetadataColumn`.
+
+    Every column is fully resolved. The enclosed :class:`~pyflink.table.types.DataType`
```

Comment:
I am curious - does the API need to be aware of unresolved version of this? It could be simpler if it did not need to know about the unresolved versions.

Reply:
I don't believe so, no - the column is just used within resolved schemas, so the column would be resolved as well. I'm pretty new to this API, what would you suggest in terms of simplification?

Reply:
I agree - I would remove "Every column is fully resolved."